timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimeLockServiceMetadataIntegrationTest.java

```diff
@@ -0,0 +1,281 @@
+            WATCHED_LOCK_2,
+            ChangeMetadata.updated("old".getBytes(StandardCharsets.UTF_8), "new".getBytes(StandardCharsets.UTF_8)),
+            WATCHED_LOCK_3,
+            ChangeMetadata.deleted("deleted".getBytes(StandardCharsets.UTF_8)),
+            WATCHED_LOCK_4,
+            ChangeMetadata.created("created".getBytes(StandardCharsets.UTF_8)));
+    private static final String UNWATCHED_TABLE_NAME = "a-random-unwatched-table";
+    private static final LockDescriptor UNWATCHED_LOCK_1 =
+            AtlasRowLockDescriptor.of(UNWATCHED_TABLE_NAME, "lock1".getBytes(StandardCharsets.UTF_8));
+    private static final IdentifiedLockRequest WATCHED_LOCK_REQUEST_WITH_METADATA =
+            standardRequestWithMetadata(ALL_WATCHED_LOCKS_WITH_METADATA);
+
+    private final LockLog lockLog = new LockLog(new MetricRegistry(), () -> 10000L);
+    private final ScheduledExecutorService scheduledExecutorService = PTExecutors.newSingleThreadScheduledExecutor();
+    private final AsyncLockService asyncLockService =
+            AsyncLockService.createDefault(lockLog, scheduledExecutorService, scheduledExecutorService);
+    private final AsyncTimelockServiceImpl timeLockService =
+            new AsyncTimelockServiceImpl(asyncLockService, new InMemoryTimestampService(), lockLog);
+    private final ConjureStartTransactionsRequest startTransactionsRequestWithInitialVersion =
+            ConjureStartTransactionsRequest.builder()
+                    .requestId(UUID.randomUUID())
+                    .lastKnownVersion(ConjureIdentifiedVersion.of(
+                            asyncLockService.leaderTime().id().id(), 0))
+                    .numTransactions(1)
+                    .requestorId(UUID.randomUUID())
+                    .build();
+
+    @Before
+    public void setupLockServiceWithWatches() {
+        timeLockService.startWatching(
+                LockWatchRequest.of(ImmutableSet.of(LockWatchReferences.entireTable(WATCHED_TABLE_NAME))));
+    }
+
+    @Test
+    public void lockEventContainsMetadataFromRequestIfWatched() {
+        waitForFuture(timeLockService.lock(WATCHED_LOCK_REQUEST_WITH_METADATA));
+
+        assertThat(getAllLockEventsMetadata())
+                .containsExactly(Optional.of(LockRequestMetadata.of(ALL_WATCHED_LOCKS_WITH_METADATA)));
+    }
+
+    @Test
+    public void lockEventDoesNotContainMetadataThatIsNotWatched() {
+        IdentifiedLockRequest mixedRequest =
+                standardRequestWithMetadata(ImmutableMap.<LockDescriptor, ChangeMetadata>builder()
+                        .putAll(ALL_WATCHED_LOCKS_WITH_METADATA)
+                        .put(
+                                UNWATCHED_LOCK_1,
+                                ChangeMetadata.updated(
+                                        "bla".getBytes(StandardCharsets.UTF_8),
+                                        "blabla".getBytes(StandardCharsets.UTF_8)))
+                        .buildKeepingLast());
```

Comment:
why do you need `buildKeepingLast`? I'd use `build` as I think all the keys here are unique, so that way you know when you're shooting yourself in the foot :)

Reply:
`build` does not compile (emits a warning), I think you mean `buildOrThrow` (I have changed it to this)